src/simp.c

```diff
@@ -1187,6 +1187,9 @@ static tree_t simp_cond_assign(tree_t t)
    else {
       tree_t s = tree_new(T_IF);
       tree_set_loc(s, tree_loc(t));
+      ident_t label = tree_ident(t);
+      if (label != NULL)
```

Comment:
Use `tree_has_ident` to check if the ident is null or not. On a debug build `tree_ident` will assert if it's null so you probably want to configure with `--enable-debug --disable-lto` while developing.

Reply:
Done, thanks, I will try with the debug build.